packages/framework/src/Model/AdvancedSearchOrder/Filter/OrderWithdrawalRequestFilter.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Model\AdvancedSearchOrder\Filter;
+
+use Doctrine\ORM\QueryBuilder;
+use Override;
+use Shopsys\FrameworkBundle\Model\AdvancedSearch\AdvancedSearchFilterInterface;
+use Symfony\Component\Form\Extension\Core\Type\HiddenType;
+use Symfony\Component\Form\FormTypeInterface;
+
+class OrderWithdrawalRequestFilter implements AdvancedSearchFilterInterface
+{
+    public const string NAME = 'orderWithdrawalRequest';
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getName(): string
+    {
+        return self::NAME;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getAllowedOperators(): array
+    {
+        return [
+            self::OPERATOR_EXISTS,
+            self::OPERATOR_DOES_NOT_EXIST,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getValueFormType(): FormTypeInterface|string
+    {
+        return HiddenType::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function getValueFormOptions(): array
+    {
+        return [];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    #[Override]
+    public function extendQueryBuilder(QueryBuilder $queryBuilder, array $rulesData): void
+    {
+        foreach ($rulesData as $ruleData) {
+            if ($ruleData->operator === self::OPERATOR_EXISTS) {
+                $queryBuilder->andWhere('o.withdrawalRequest IS NOT NULL');
```

Comment:
I would like to discuss one more thing - the owning side of the 1:1 relation.
I have noticed we have foreign key on both sides (order and withdrawal_request). 

This is brittle (on conflict, what is the source of truth?), cost more on insert and it's not even possible to delete a withdrawal (ok, that is not probably needed anyway). 

I would keep the FK only in the withdrawal_requests table with onDelete cascade set (because that prevent existence of the "orphaned" withdrawals)
(1:1 relation is enforced by unique index on the order_id column)